x/cdp/keeper/auctions.go

```diff
@@ -0,0 +1,233 @@
+func (k Keeper) AuctionCollateral(ctx sdk.Context, deposits types.Deposits, debt sdk.Int, bidDenom string) sdk.Error {
+	auctionSize := k.getAuctionSize(ctx, deposits[0].Amount[0].Denom)
+	partialAuctionDeposits := partialDeposits{}
+	totalCollateral := deposits.SumCollateral()
+	for totalCollateral.GT(sdk.ZeroInt()) {
+		for i, dep := range deposits {
+			// create auctions from individual deposits that are larger than the auction size
+			debtChange, collateralChange, err := k.CreateAuctionsFromDeposit(ctx, dep, debt, totalCollateral, auctionSize, bidDenom)
+			if err != nil {
+				return err
+			}
+			debt = debt.Sub(debtChange)
+			totalCollateral = totalCollateral.Sub(collateralChange)
+			dep.Amount = sdk.NewCoins(sdk.NewCoin(dep.Amount[0].Denom, dep.Amount[0].Amount.Sub(collateralChange)))
+			// if there is leftover collateral that is less than a lot
+			if !dep.Amount.IsZero() {
+				collateralAmount := dep.Amount[0].Amount
+				collateralDenom := dep.Amount[0].Denom
+				// figure out how much debt this deposit accounts for
+				// (depositCollateral / totalCollateral) * totalDebtFromCDP
+				debtCoveredByDeposit := (collateralAmount.Quo(totalCollateral)).Mul(debt)
+				// if adding this deposit to the other partial deposits is less than a lot
+				if (partialAuctionDeposits.SumCollateral().Add(collateralAmount)).LT(auctionSize) {
+					// append the deposit to the partial deposits and zero out the deposit
+					pd := newPartialDeposit(dep.Depositor, dep.Amount, debtCoveredByDeposit)
+					partialAuctionDeposits = append(partialAuctionDeposits, pd)
+					dep.Amount = sdk.NewCoins(sdk.NewCoin(collateralDenom, sdk.ZeroInt()))
+				} else {
+					// if the sum of partial deposits now makes a lot
+					partialCollateral := sdk.NewCoins(sdk.NewCoin(collateralDenom, auctionSize.Sub(partialAuctionDeposits.SumCollateral())))
+					partialAmount := partialCollateral[0].Amount
+					partialDebt := (partialAmount.Quo(collateralAmount)).Mul(debtCoveredByDeposit)
+
+					// create a partial deposit from the deposit
+					partialDep := newPartialDeposit(dep.Depositor, partialCollateral, partialDebt)
+					// append it to the partial deposits
+					partialAuctionDeposits = append(partialAuctionDeposits, partialDep)
+					// create an auction from the partial deposits
+					debtChange, collateralChange, err := k.CreateAuctionFromPartialDeposits(ctx, partialAuctionDeposits, debt, totalCollateral, auctionSize, bidDenom)
+					if err != nil {
+						return err
+					}
+					debt = debt.Sub(debtChange)
+					totalCollateral = totalCollateral.Sub(collateralChange)
+					// reset partial deposits and update the deposit amount
+					partialAuctionDeposits = partialDeposits{}
+					dep.Amount = sdk.NewCoins(sdk.NewCoin(collateralDenom, collateralAmount.Sub(partialAmount)))
+				}
+			}
+			if dep.Amount.IsZero() {
+				// remove the deposit from the slice if it is empty
+				deposits = append(deposits[:i], deposits[i+1:]...)
+				i--
```

Comment:
my editor is saying `i--` is an ineffective assignment. Maye although it's updated here, the next for loop iteration overwrites it?
Could also be a false positive